src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoEnchanter.java

```diff
@@ -50,8 +54,14 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
 
                 for (Map.Entry<Enchantment, Integer> e : meta.getStoredEnchants().entrySet()) {
                     if (e.getKey().canEnchantItem(target)) {
-                        amount++;
-                        enchantments.put(e.getKey(), e.getValue());
+                        if (e.getValue() <= enchantLevelLimit.getValue() || enchantLevelLimit.getValue() < 1) {
+                            amount++;
+                            enchantments.put(e.getKey(), e.getValue());
+                        } else {
+                            if (!menu.toInventory().getViewers().isEmpty()) {
+                                menu.toInventory().getViewers().forEach(p -> SlimefunPlugin.getLocalization().sendMessage(p, "messages.above-limit-level", true, (message -> message.replace("%level%", enchantLevelLimit.getValue().toString()))));
```

Comment:
1. Never use `.forEach(Consumer)`.
2. Sending a message to all inventory viewers is rather akwards, not something that we do here. I reckon it would make more sense to utilize the "progress bar" slot in the middle for this and update that instead.